src/databricks/sql/client.py

```
@@ -456,7 +456,16 @@ def __enter__(self) -> "Cursor":
         return self
 
     def __exit__(self, exc_type, exc_value, traceback):
-        self.close()
+        try:
```

The default behaviour is anyway to not suppress an error. The `False` is just redundant. The logging is useful but catching exception and then throwing it again is the default behaviour and explicit handling don't think adds anything valuable.